src/wing_segmenter/cli.py

```diff
@@ -0,0 +1,125 @@
+    segment_parser.add_argument('--crop-by-class', action='store_true',
+                                help='Enable cropping of segmented classes into crops/ directory.')
+    segment_parser.add_argument('--force', action='store_true',
+                                help='Force reprocessing even if outputs already exist.')
+
+    # Background removal options
+    bg_group = segment_parser.add_argument_group('Background Removal Options')
+    bg_group.add_argument('--remove-crops-background', action='store_true',
+                           help='Remove background from cropped images.')
+    bg_group.add_argument('--remove-full-background', action='store_true',
+                           help='Remove background from the entire (resized or original) image.')
+    bg_group.add_argument('--background-color', choices=['white', 'black'], default=None,
+                           help='Background color to use when removing background.')
+
+    # Subcommand: scan-runs
+    scan_parser = subparsers.add_parser('scan-runs', help='List existing processing runs for a dataset.')
+    scan_parser.add_argument('--dataset', required=True, help='Path to the dataset directory.')
+    scan_parser.add_argument('--output-dir', default=None, help='Base path where outputs were stored.')
+
+    # Parse arguments
+    args = parser.parse_args()
+
+    # Command input validations
+    if args.command == 'segment':
+        # If size is provided, enforce resizing options
+        if args.size:
+            if len(args.size) not in [1, 2]:
+                parser.error('--size must accept either one value (square resize) or two values (width and height).')
+            if not args.resize_mode:
+                parser.error('--resize-mode must be specified when --size is provided.')
+        # If no size is provided, ensure that resizing options were not explicitly set
+        else:
+            if args.resize_mode is not None:
+                parser.error('Resizing options (--resize-mode) require --size to be specified.')
+            if args.padding_color is not None:
+                parser.error('Resizing options (--padding-color) require --size to be specified.')
+
+        # --remove-crops-background requires --crop-by-class
+        if args.remove_crops_background and not args.crop_by_class:
+            parser.error('--remove-crops-background requires --crop-by-class to be set.')
+
+        # Need to set croped or full background removal to set background color
+        if args.background_color and not (args.remove_crops_background or args.remove_full_background):
+            parser.error('--background-color can only be set when background removal is enabled.')
+
+        # Ensure that if --custom-output-dir is set, --outputs-base-dir is not used
+        if args.custom_output_dir and args.outputs_base_dir:
+            parser.error('Cannot specify both --outputs-base-dir and --custom-output-dir. Choose one.')
```

Comment:
Is this not handled by the [mutually exclusive group](https://docs.python.org/3/library/argparse.html#mutual-exclusion)?